src/bin/ion/commands/beta/generate/generator.rs

```diff
@@ -213,245 +332,20 @@ impl<'a> CodeGenerator<'a> {
     ///   }
     /// }
     /// ```
-    /// For the above schema, both `fields` and `type` constraints map to different data models
+    /// For the above schema, both `fields` and `type` constraints map to different abstract data types
     /// respectively Struct(with given fields `source` and `destination`) and Value(with a single field that has String data type).
-    fn verify_data_model_consistency(
+    fn verify_abstract_data_type_consistency(
         &mut self,
-        current_data_model: DataModel,
+        current_abstract_data_type: AbstractDataType,
         code_gen_context: &mut CodeGenContext,
     ) -> CodeGenResult<()> {
-        if let Some(data_model) = &code_gen_context.data_model {
-            if data_model != &current_data_model {
-                return invalid_data_model_error(format!("Can not determine abstract data type as current constraint {} conflicts with prior constraints for {}.", current_data_model, data_model));
+        if let Some(abstract_data_type) = &code_gen_context.abstract_data_type {
+            if abstract_data_type != &current_abstract_data_type {
+                return invalid_abstract_data_type_error(format!("Can not determine abstract data type as current constraint {} conflicts with prior constraints for {}.", current_abstract_data_type, abstract_data_type));
             }
         } else {
-            code_gen_context.with_data_model(current_data_model);
+            code_gen_context.with_abstract_data_type(current_abstract_data_type);
         }
         Ok(())
     }
-
-    /// Generates a struct field based on field name and value(data type)
-    fn generate_struct_field(
-        &mut self,
-        tera_fields: &mut Vec<Field>,
-        isl_type_ref: &IslTypeRef,
-        modules: &mut Vec<String>,
-        field_name: &str,
-        imports: &mut Vec<Import>,
-        code_gen_context: &mut CodeGenContext,
-    ) -> CodeGenResult<()> {
-        let value = self.generate_field_value(isl_type_ref, modules, imports, code_gen_context)?;
-
-        tera_fields.push(Field {
-            name: {
-                match self.language {
-                    Language::Rust => field_name.to_case(Case::Snake),
-                    Language::Java => field_name.to_case(Case::Camel),
-                }
-            },
-            value,
-        });
-        Ok(())
-    }
-
-    /// Generates field value in a struct which represents a data type in codegen's programming language
-    fn generate_field_value(
-        &mut self,
-        isl_type_ref: &IslTypeRef,
-        modules: &mut Vec<String>,
-        imports: &mut Vec<Import>,
-        code_gen_context: &mut CodeGenContext,
-    ) -> CodeGenResult<String> {
-        Ok(match isl_type_ref {
-            IslTypeRef::Named(name, _) => {
```

Comment:
(🗺️ PR tour) `type_reference_name` has been added that stores the type reference name and hence this method has been removed.